docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -0,0 +1,236 @@
+# Shoot Control Plane Migration
+
+## Motivation
+
+Currently moving the control plane of a shoot cluster can only be done manually and requires deep knowledge of how exactly to transfer the resources and state from one seed to another. This can make it slow and prone to errors.
+
+Automatic migration can be very useful in a couple of scenarios:
+- Seed goes down and can't be repaired (fast enough or at all) and it's control planes need to be brought to another seed
+- Seed needs to be changed, but this operation requires the recreation of the seed (e.g. turn a single-AZ seed into a multi-AZ seed)
+- Seeds need to be rebalanced
+- New seeds became available in a region closer to/in the region of the workers and the control plane should be moved there to improve latency
+- Gardener ring, which is a self-supporting setup/underlay for a highly available (usually cross-region) Gardener deployment
+
+
+## Goals
+
+* Provide a mechanism to migrate the control plane of a shoot cluster from one seed to another
+* The mechanism should support migration from a seed which is no longer reachable (Disaster Recovery)
+* The mechanism can automatically decide whether a seed is unhealthy and control planes need to be migrated to a new seed.
+* The mechanism needs to take into account that gardenlet will be used for shoot reconciliation very soon
+* The shoot cluster nodes are preserved and continue to run the workload, but will talk to the new control plane after the migration completes
+* Extension controllers implement a mechanism which allows them to store their state or to be restored from an already existing state on a different seed cluster.
+* The already existing shoot reconciliation flow is reused for migration with minimum changes
+
+
+## Terminology
+
+__Source Seed__ is the seed which currently hosts the control plane of a Shoot Cluster
+
+__Destination Seed__ is the seed to which the control plane is being migrated
+
+
+## Resources and controller state which have to be migrated between two seeds:
+
+**Note:** The following lists are just FYI and are meant to show the current resources which need to be moved to the __Destination Seed__
+
+### Secrets
+
+Gardener has preconfigured lists of needed secrets which are generated when a shoot is created and deployed in the seed. Following is a minimum set of secrets which must be migrated to the New Seed. Other secrets can be regenerated from them.
+
+- ca
+- ca-front-proxy
+- static-token
+- ca-kubelet
+- ca-metrics-server
+- etcd-encryption-secret
+- kube-aggregator
+- kube-apiserver-basic-auth
+- kube-apiserver
```

Comment:
This one is not stronlgy needed, I guess. Or do I miss something?

Reply:
We moved it during the manual migrations we did. Let's have another check if we can skip it and then I'll remove it from the list. (same for the next comment)